akka-actor/src/main/scala/akka/io/dns/DnsSettings.scala

```diff
@@ -25,20 +25,22 @@ private[dns] final class DnsSettings(system: ExtendedActorSystem, c: Config) {
   import DnsSettings._
 
   val NameServers: List[InetSocketAddress] = {
-    val addresses = Try(c.getString("nameservers")).toOption match {
-      case Some("default") ⇒
-        val osAddresses = getDefaultNameServers(system).getOrElse(failUnableToDetermineDefaultNameservers)
-        if (osAddresses.isEmpty) failUnableToDetermineDefaultNameservers
-        osAddresses
-      case Some(other) ⇒
-        parseNameserverAddress(other) :: Nil
-      case None ⇒
+    c.getValue("nameservers").valueType() match {
```

Comment:
Meh, I was hoping there was a nicer way to do this, without having to read the same config multiple times, something like:
```
c.getValue("nameservers") match {
  case c if c.valueType() == STRING && c.value() == "default" => yadi
  case c if c.valueType() == STRING => yada
}
```

But I must have imagined that something like that was possible. Not worth spending more time on though.